src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -122,43 +74,41 @@ internal static bool DoSslHandshake(IntPtr sslContextPtr, IntPtr recvPtr, int re
 
             if (retVal != 1)
             {
-                error = GetSslError(context.sslPtr, retVal);
+                error = GetSslError(context, retVal);
 
                 if ((retVal != -1) || (error != libssl.SslErrorCode.SSL_ERROR_WANT_READ))
                 {
-                    throw CreateSslException(context.sslPtr, "SSL Handshake failed: ", retVal);
+                    throw CreateSslException(context, "SSL Handshake failed: ", retVal);
                 }
             }
```

Comment:
Rather than checking retVal twice, could you just do:
```C#
int retVal = libssl.SSL_do_handshake(context);
if (retVal == 1)
{
    if (!context.IsServer)
        return true;
}
else
{
    int error = GetSslError(context, retVal);
    ... // etc.
}
```
?